### PR TITLE
W2 -tienhuynh3-Create calculator view

### PR DESCRIPTION
**Pull request by at-tienhuynh3**
#### Related issues (Optional)
Task #14 

#### Description (Required)
- [x] Create CaculatorAcivity
- [x] Make Caculation

### CODE REVIEW

File: app/src/at-tienhuynh/java/vn/asiantech/internship/Caculation.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * Created at 2017
+ * Created by jackty on 27/11/2017.
+ */
+public class Caculation {
+    private float mResult;
+
+    public Caculation() {
+    }
+
+    protected Float Sum(float numA, float numB, TextView tvResult) {
```
Comment (at-dev-support-01) [inline app/src/at-tienhuynh/java/vn/asiantech/internship/Caculation.java]: ERROR: [MethodNamingConventions] Method names should not start with capital letters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#MethodNamingConventions

File: app/src/at-tienhuynh/java/vn/asiantech/internship/Caculation.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        mResult = numA + numB;
+        tvResult.setText(String.valueOf(mResult));
+        return mResult;
+    }
+
+    protected Float Subtraction(float numA, float numB, TextView tvResult) {
```
Comment (at-dev-support-01) [inline app/src/at-tienhuynh/java/vn/asiantech/internship/Caculation.java]: ERROR: [MethodNamingConventions] Method names should not start with capital letters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#MethodNamingConventions

File: app/src/at-tienhuynh/java/vn/asiantech/internship/Caculation.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        mResult = numA - numB;
+        tvResult.setText(String.valueOf(mResult));
+        return mResult;
+    }
+
+    protected Float Multiplication(float numA, float numB, TextView tvResult) {
```
Comment (at-dev-support-01) [inline app/src/at-tienhuynh/java/vn/asiantech/internship/Caculation.java]: ERROR: [MethodNamingConventions] Method names should not start with capital letters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#MethodNamingConventions

File: app/src/at-tienhuynh/java/vn/asiantech/internship/Caculation.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        mResult = numA * numB;
+        tvResult.setText(String.valueOf(mResult));
+        return mResult;
+    }
+
+    protected Float Division(float numA, float numB, TextView tvResult) {
```
Comment (at-dev-support-01) [inline app/src/at-tienhuynh/java/vn/asiantech/internship/Caculation.java]: ERROR: [MethodNamingConventions] Method names should not start with capital letters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#MethodNamingConventions

File: app/src/at-tienhuynh/res/layout/activity_caculator_.xml [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    </LinearLayout>
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="10dp"
+        android:orientation="horizontal"
+        android:weightSum="4">
+
+        <Button
```
Comment (at-dev-support-01) [inline app/src/at-tienhuynh/res/layout/activity_caculator_.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-tienhuynh/res/layout/activity_caculator_.xml [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_margin="5dp"
+            android:layout_weight="1"
+            android:gravity="center"
+            android:text="@string/edtSum"
+            android:textSize="15dp"
+            android:textStyle="bold" />
+
+        <Button
```
Comment (at-dev-support-01) [inline app/src/at-tienhuynh/res/layout/activity_caculator_.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-tienhuynh/res/layout/activity_caculator_.xml [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_margin="5dp"
+            android:layout_weight="1"
+            android:gravity="center"
+            android:text="@string/edtSub"
+            android:textSize="15dp"
+            android:textStyle="bold" />
+
+        <Button
```
Comment (at-dev-support-01) [inline app/src/at-tienhuynh/res/layout/activity_caculator_.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-tienhuynh/res/layout/activity_caculator_.xml [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_margin="5dp"
+            android:layout_weight="1"
+            android:gravity="center"
+            android:text="@string/edtMulti"
+            android:textSize="15dp"
+            android:textStyle="bold" />
+
+        <Button
```
Comment (at-dev-support-01) [inline app/src/at-tienhuynh/res/layout/activity_caculator_.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-tienhuynh/res/layout/activity_caculator_.xml [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+    </LinearLayout>
+
+    <TextView
+        android:id="@+id/tvResult"
+        android:layout_width="match_parent"
+        android:layout_height="50dp"
+        android:layout_margin="10dp"
+        android:background="@color/colorAccent"
+        android:gravity="center"
+        android:text="Result"
```
Comment (at-dev-support-01) [inline app/src/at-tienhuynh/res/layout/activity_caculator_.xml]: WARNING: [HardcodedText] Hardcoded string "Result", should use `@string` resource
 Hardcoding text attributes directly in layout files is bad for several reasons:

* When creating configuration variations (for example for landscape or portrait)you have to repeat the actual text (and keep it up to date when making changes)

* The application cannot be translated to other languages by just adding new translations for existing string resources.

There are quickfixes to automatically extract this hardcoded string into a resource lookup.

**Comment by at-tienhuynh3** [comment]
@at-huongnguyen Please review for me. Thank you

**Comment by at-huongnguyen** [comment]
@at-tienhuynh3 
Please resolve conflicts

**Comment by at-tienhuynh3** [comment]
@at-huongnguyen i fixed conflicts, please review for me , thank you!

File: app/build.gradle [hunk 10]
```diff
@@ -81,6 +81,7 @@ dependencies {
     implementation 'com.android.support:appcompat-v7:26.1.0'
     implementation 'com.android.support:recyclerview-v7:26.1.0'
     implementation 'com.android.support:cardview-v7:26.1.0'
+    implementation 'com.android.support.constraint:constraint-layout:1.0.2'
```
Comment (at-huongnguyen) [inline app/build.gradle]: Remove `constraint-layout`

File: app/src/at-tienhuynh/java/vn/asiantech/internship/calculation/CaculatorActivity.java [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+    private EditText mEdtNumberA;
+    private EditText mEdtNumberB;
+    private Button mBtnSum;
+    private Button mBtnSubtraction;
+    private Button mBtnMultiplication;
+    private Button mBtnDivision;
+    private TextView mTvResult;
+    private TextView mTvCalculation;
+    private Calculation mCalculation;
+    private Boolean mIsEmty;
```
Comment (at-huongnguyen) [inline app/src/at-tienhuynh/java/vn/asiantech/internship/calculation/CaculatorActivity.java]: Use: boolean

File: app/src/at-tienhuynh/java/vn/asiantech/internship/calculation/CaculatorActivity.java [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+        mBtnSubtraction = findViewById(R.id.btnSubtraction);
+        mBtnMultiplication = findViewById(R.id.btnMultiplication);
+        mBtnDivision = findViewById(R.id.btnDivision);
+        mTvResult = findViewById(R.id.tvResult);
+        mTvCalculation = findViewById(R.id.tvCalculation);
+        mCalculation = new Calculation();
+        mBtnSum.setOnClickListener(this);
+        mBtnSubtraction.setOnClickListener(this);
+        mBtnMultiplication.setOnClickListener(this);
+        mBtnDivision.setOnClickListener(this);
```
Comment (at-huongnguyen) [inline app/src/at-tienhuynh/java/vn/asiantech/internship/calculation/CaculatorActivity.java]: Create initListener() method

File: app/src/at-tienhuynh/java/vn/asiantech/internship/calculation/CaculatorActivity.java [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+                    break;
+                }
+        }
+    }
+
+    private void getCalculator(Button btnCalculator) {
+        mTvCalculation.setText(btnCalculator.getText().toString());
+    }
+
+    private void checkEmpty() {
```
Comment (at-huongnguyen) [inline app/src/at-tienhuynh/java/vn/asiantech/internship/calculation/CaculatorActivity.java]: optimize this function

File: app/src/at-tienhuynh/java/vn/asiantech/internship/calculation/Calculation.java [hunk 14]
```diff
@@ -0,0 +1,34 @@
+package vn.asiantech.internship.calculation;
+
+import android.widget.TextView;
+
+/**
+ * Created at 2017
+ * Created by jackty on 27/11/2017.
+ */
+class Calculation {
+    private float mResult;
```
Comment (at-huongnguyen) [inline app/src/at-tienhuynh/java/vn/asiantech/internship/calculation/Calculation.java]: Rename `mResult` -> `result`

File: app/src/at-tienhuynh/res/layout/activity_caculator.xml [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+            android:inputType="number"
+            android:maxLines="1" />
+    </LinearLayout>
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="10dp"
+        android:orientation="horizontal"
+        android:weightSum="4">
```
Comment (at-huongnguyen) [inline app/src/at-tienhuynh/res/layout/activity_caculator.xml]: Remove weightSum

File: app/src/at-tienhuynh/res/layout/activity_caculator.xml [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+    tools:context="vn.asiantech.internship.calculation.CaculatorActivity">
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_gravity="center"
+        android:layout_marginTop="10dp"
+        android:gravity="center"
+        android:orientation="horizontal"
+        android:weightSum="11">
```
Comment (at-huongnguyen) [inline app/src/at-tienhuynh/res/layout/activity_caculator.xml]: Remove weightSum

File: app/src/at-tienhuynh/java/vn/asiantech/internship/SplashActivity.java [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_splash);
+        initViews();
+    }
+
+    private void initViews() {
+        mBtnLoginScreen = findViewById(R.id.btnLoginScreen);
+        mBtnCalculationScreen = findViewById(R.id.btnCalculatorScreen);
+        mBtnLoginScreen.setOnClickListener(this);
+        mBtnCalculationScreen.setOnClickListener(this);
```
Comment (at-huongnguyen) [inline app/src/at-tienhuynh/java/vn/asiantech/internship/SplashActivity.java]: Create initListener method

File: app/src/at-tienhuynh/java/vn/asiantech/internship/MainActivity.java [hunk 18]
```diff
@@ -0,0 +1,13 @@
+package vn.asiantech.internship;
```
Comment (at-huongnguyen) [inline app/src/at-tienhuynh/java/vn/asiantech/internship/MainActivity.java]: If MainActivity.class not use, plz remove it

File: app/src/at-tienhuynh/java/vn/asiantech/internship/SplashActivity.java [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
+
+import vn.asiantech.internship.calculation.CaculatorActivity;
+import vn.asiantech.internship.login.LoginActivity;
+
+public class SplashActivity extends AppCompatActivity implements View.OnClickListener {
+    private Button mBtnLoginScreen;
+    private Button mBtnCalculationScreen;
```
Comment (at-huongnguyen) [inline app/src/at-tienhuynh/java/vn/asiantech/internship/SplashActivity.java]: Move to local variable

**Comment by at-tienhuynh3** [comment]
i finished fix your comment , plz review again for me @at-huongnguyen

**Pull request merged**